lib/index.js

```diff
@@ -1723,38 +1749,33 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
+        var obj = x;  // variable used to transverse prototype chain
+        function testAndAdd (prop) {
+            if (!(prop in out) && f(prop, x[prop])) {
```

Comment:
This would result in overridden properties being passed to `f`. Example

```javascript
function f(p, v) {
    return v === 1;
}

var obj1 = {v: 1};
var obj2 = Object.create(obj1);
obj2.v = 2;

_([obj2]).pickBy(f).toArray(_.log) // => [{v: 1}]
```

In the reverse, it'll also disallow picking on default properties of objects, like `hasOwnProperty`, but that's not as important.